src/Shared/Microsoft.DotNet.Internal.Testing.Utility/StringRegexAssertionExtensions.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.RegularExpressions;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using FluentAssertions.Primitives;
+
+namespace Microsoft.DotNet.Internal.Testing.Utility
+{
+    public static class StringRegexAssertionExtensions{
+        public static AndConstraint<StringAssertions> MatchRegex(
+            this StringAssertions stringAssertion,
+            Regex regularExpression,
+            string because = "",
+            params object[] becauseArgs
+        )
+        {
+            Execute.Assertion.ForCondition(stringAssertion.Subject != null)
+                .UsingLineBreaks.BecauseOf(because, becauseArgs)
+                .FailWith("Expected {context:string} to match regex {0}{reason}, but it was <null>.",
+                    (object) regularExpression);
```

Comment:
>(object) regularExpression [](start = 20, length = 26)

Do regexes have a good ToString?

Reply:
Yup, it just returns the actual string. It doesn't include the options, but that's fine.

Reply:
Made in a... roundabout way...